cosmos/scheduler/cmd/scheduler.go

```diff
@@ -0,0 +1,247 @@
+package scheduler
+
+import (
+	"context"
+	"log"
+
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	"github.com/scality/zenko/cosmos/api/types/v1alpha1"
+	clientV1alpha1 "github.com/scality/zenko/cosmos/clientset/v1alpha1"
+	"github.com/scality/zenko/cosmos/scheduler/pkg"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Scheduler represents a Cosmos Scheduler instance
+type Scheduler struct {
+	KubeAlpha     *clientV1alpha1.ExampleV1Alpha1Client
+	KubeClientset *kubernetes.Clientset
+	Pensieve      *pensieve.Helper
+	Namespace     string
+	Cloudserver   string
+	MongodbClient *mongo.Client
+}
+
+// Run starts the Cosmos Scheduler
+func (s *Scheduler) Run() {
+	log.Println("starting cosmos scheduler")
+	exists, err := s.getIngestionSecret()
+	if err != nil {
+		panic(err.Error())
+	}
+	if !exists {
+		if err = s.createIngestionSecret(); err != nil {
+			panic(err.Error())
+		}
+	}
+	msgCh := make(chan interface{}, 1)
+	go func() {
+		for {
+			s.handler(msgCh)
+		}
+	}()
+	for {
+		if err := s.waitForOverlayUpdate(); err != nil {
+			log.Println(err)
+		} else {
+			msgCh <- "overlay update"
+		}
+	}
+}
+
+func (s *Scheduler) waitForOverlayUpdate() error {
```

Comment:
I think `waitForOverlayUpdate` is creating a new `changeStream` for each  it's called.
maybe move the loop logic into the method?

an example using `sync.WaitGroup`:

```
func (s *Scheduler) Run() {
          ...
          wg := sync.WaitGroup
          defer wg.Wait()
          wg.Add(1)
          ... handler(msgCh, &wg) goroutine
          wg.Add(1)
          go waitForOverlayUpdate(msgCh, &wg)
          ...
}

func (s *Schedule) waitForOverlayUpdate(ch chan<- string, wg *sync.WaitGroup) {
        collection := s.MongodbClient.Database("metadata").Collection("PENSIEVE")
	ctx := context.Background()
	cur, err := collection.Watch(ctx, mongo.Pipeline{})
        defer cur.Close(ctx)
	if err != nil {
		return err
	}
        for {
                if cur.Next(ctx) {
                        ch <- "update overlay"
                } else {
                        if err := cur.Err(); err != nil {
                                log.Println(err)
                        }
                }
        }
        wg.Done()
}
```